### ISSUE TITLE
can not upload image with curl

### ISSUE THREAD

**Issue by jiturawat94** [ixudra/curl#50]
im using this
$response = Curl::to($to)
->withData( $data)
->withContentType('multipart/form-data')
->containsFile()
->post();

and when im trying to return this from method of api controller
$request->paymentProof;

its returning a string
'C:\wamp64\tmp\php510E.tmp'

**Comment by gifary** [comment]
Hello jituwat,
I have same problem with you and I solve with this function 

```
private function getCurlValue($filename, $contentType, $postname)
    {
        // PHP 5.5 introduced a CurlFile object that deprecates the old @filename syntax
        // See: https://wiki.php.net/rfc/curl-file-upload
        if (function_exists('curl_file_create')) {
            return curl_file_create($filename, $contentType, $postname);
        }
    
        // Use the old style if using an older version of PHP
        $value = "@{$filename};filename=" . $postname;
        if ($contentType) {
            $value .= ';type=' . $contentType;
        }
    
        return $value;
    }
```
How to use that function.
 ```
        $file_path = public_path('storage/hello.txt');
        $cfile = $this->getCurlValue($file_path,false,'tes.txt');

        $response = Curl::to($this->base_url."loker")
            ->withData(array("foto"=>$cfile))
            ->withContentType('multipart/form-data')
            ->withHeader('Authorization: Bearer '.$this->access_token)
            ->withHeader('Accept: application/json')
            ->containsFile()
            ->post();
```
Maybe this can help you.

**Comment by elimentz** [comment]
Bug has been addressed in v 6.12.0. The suggestion provided by @gifary is will now happen automatically when using the `withFile()` method. See documentation for details:

For sending files via a POST request, you can use the `withFile` method to correctly format a request before sending:

```php

    use Ixudra\Curl\Facades\Curl;

    $response = Curl::to('http://foo.com/bar')
        ->withData( array( 'Foo' => 'Bar' ) )
        ->withFile( 'image_1', '/path/to/dir/image1.png', 'image/png', 'imageName1.png' )
        ->withFile( 'image_2', '/path/to/dir/image2.png', 'image/png', 'imageName2.png' )
        ->post();

```

You can add as many files to the request as you want. A couple of things to keep in mind:

- When submitting files, the `asJson()` method and `asJsonRequest()` method cannot be used. If you do, the files will not be transferred correctly
- The files are added to the data that was provided in the `withData()` method using the first parameter of the `withFile()` method. If this key already exists, it will be overridden.

Thanks for reporting and apologies for the delay
